pkg/packagekit/kpatch.jsx

```diff
@@ -0,0 +1,371 @@
+                const current = this.current(this.kpatchService.enabled, state.patchInstalled, state.patchUnavailable);
+                return ({
+                    enabled: this.kpatchService.enabled,
+                    justCurrent: current && !state.auto,
+                    applyCheckbox: current,
+                });
+            });
+        });
+    }
+
+    checkSetup() {
+        // TODO - replace both with `dnf kpatch status` once https://github.com/dynup/kpatch-dnf/pull/8 lands
+        const kpatch_promise = cockpit.file("/etc/dnf/plugins/kpatch.conf").read()
+                .then(data => {
+                    if (data) {
+                        const auto = /autoupdate\s*=\s*True/i.test(data);
+                        this.setState((state, _) => {
+                            const current = this.current(state.enabled, state.patchInstalled, state.patchUnavailable);
+                            return ({
+                                auto: !!auto,
+                                justCurrent: current && !auto,
+                                applyCheckbox: current,
+                            });
+                        });
+                    }
+                })
+                .catch(() => true); // Ignore errors, most likely just does not exist
+
+        const uname_promise = cockpit.spawn(["uname", "-r"])
+                .then(data => {
+                    const fields = data.split("-");
+                    const kpp_kernel_version = fields[0].replaceAll(".", "_");
+                    let release = fields[1].split(".");
+                    release = release.slice(0, release.length - 2); // remove el8.x86_64
+                    const kpp_kernel_release = release.join("_");
+                    const patch_name = ["kpatch-patch", kpp_kernel_version, kpp_kernel_release].join("-");
+                    return check_missing_packages([patch_name])
+                            .then(d =>
+                                this.setState((state, _) => {
+                                    const installed = (d.unavailable_names || []).length === 0 && (d.missing_names || []).length === 0;
+                                    const unavailable = (d.unavailable_names || []).length > 0;
+                                    const current = this.current(state.enabled, installed, unavailable);
+                                    return ({
+                                        kernelName: data,
+                                        patchName: patch_name,
+                                        patchInstalled: installed,
+                                        patchUnavailable: unavailable,
+                                        justCurrent: current && !state.auto,
+                                        applyCheckbox: current,
+                                    });
+                                })
+                            );
+                })
+                .catch(console.error);
```

Comment:
Does that actually work? I.e. do you get a readable string instead of '[Object]'? Might be better to drop this and leave it to the catch handler in componentDidMount to log it?

Reply:
> Does that actually work? I.e. do you get a readable string instead of '[Object]'? 

Of course, you get an object, that you can expand.

> Might be better to drop this and leave it to the catch handler in componentDidMount to log it?

Not really. It is used in different places, not just in `componentDidMount`, but mainly it is used in `Promise.allSettled` thus it will never reject in `componentDidMount`.